### PR TITLE
mon/OSDMonitor: Restrict pool names beggining with a dot

### PR DESCRIPTION
**Pull request by Matan-B**
> Pool names beginning with . are reserved for use by Ceph’s internal operations. Please do not create or manipulate pools with these names.

Restriction applies both to `ceph osd pool create` and `ceph osd pool rename`.

https://docs.ceph.com/en/latest/rados/operations/pools/#pool-names

Signed-off-by: Matan Breizman <mbreizma@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by Matan-B** [comment]
jenkins test make check

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -12829,6 +12829,13 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
 
     string poolstr;
     cmd_getval(cmdmap, "pool", poolstr);
+    bool confirm = false;
```
Comment (ronen-fr) [inline src/mon/OSDMonitor.cc:12832]: why do add the 'yes I do' requirement to this command? and if needed - why not in a separate commit?

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -12829,6 +12829,13 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
...
+    cmd_getval(cmdmap, "yes_i_really_mean_it", confirm);
+    if (poolstr[0] == '.' && !confirm) {
+      ss << "pool names begginning with . are not allowed";
```
Comment (ronen-fr) [inline src/mon/OSDMonitor.cc:12835]: spelling error

File: src/mon/OSDMonitor.cc [hunk 3]
```diff
@@ -13056,7 +13063,14 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     cmd_getval(cmdmap, "destpool", destpoolstr);
     int64_t pool_src = osdmap.lookup_pg_pool_name(srcpoolstr.c_str());
     int64_t pool_dst = osdmap.lookup_pg_pool_name(destpoolstr.c_str());
+    bool confirm = false;
```
Comment (ronen-fr) [inline src/mon/OSDMonitor.cc:13066]: same here

File: src/mon/OSDMonitor.cc [hunk 4]
```diff
@@ -13056,7 +13063,14 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
...
+    cmd_getval(cmdmap, "yes_i_really_mean_it", confirm);
 
+    if (destpoolstr[0] == '.' && !confirm) {
+      ss << "pool names begginning with . are not allowed";
```
Comment (ronen-fr) [inline src/mon/OSDMonitor.cc:13070]: same spelling error

**Review by ronen-fr — CHANGES_REQUESTED** [review]

File: src/mon/OSDMonitor.cc (hunk 1 above)
Comment (Matan-B) [reply src/mon/OSDMonitor.cc:12832]: `mgr_module.py` uses cli commands to create it's pool. Extra flag must be used (when creating `.mgr` pool) in order to differentiate between our internal creations and the rest.

File: src/mon/OSDMonitor.cc (hunk 2 above)
Comment (Matan-B) [reply src/mon/OSDMonitor.cc]: Fixed, Thanks!

File: src/mon/OSDMonitor.cc (hunk 1 above)
Comment (Matan-B) [reply src/mon/OSDMonitor.cc:12832]: Comment was added for clarification.

File: src/mon/MonCommands.h [hunk 5]
```diff
@@ -1061,7 +1061,8 @@ COMMAND("osd pool create "
 	"name=autoscale_mode,type=CephChoices,strings=on|off|warn,req=false "
 	"name=bulk,type=CephBool,req=false "
 	"name=target_size_bytes,type=CephInt,range=0,req=false "
-	"name=target_size_ratio,type=CephFloat,range=0|1,req=false",\
+	"name=target_size_ratio,type=CephFloat,range=0|1,req=false "\
+	"name=yes_i_really_mean_it,type=CephBool,req=false",
```
Comment (ronen-fr) [inline src/mon/MonCommands.h:1065]: isn't there a problem with the order of the lines? note the '\' that makes the "create pool" part of the previous line

**Review by ronen-fr — CHANGES_REQUESTED** [review]
almost finished. See one comment.

File: src/mon/MonCommands.h (hunk 5 above)
Comment (ronen-fr) [reply src/mon/MonCommands.h:1065]: No - actually there's an unneeded '\' in the previous line

**Review by ronen-fr — APPROVED** [review]

**Comment by neha-ojha** [comment]
``` 
2022-02-11T19:05:03.319 INFO:journalctl@ceph.mon.a.smithi164.stdout:Feb 11 19:05:03 smithi164 bash[12809]: cluster 2022-02-11T19:05:02.244956+0000 mgr.a (mgr.14302) 10 : cluster [DBG] pgmap v6: 1 pgs: 1 active+clean; 577 KiB data, 17 MiB used, 268 GiB / 268 GiB avail
2022-02-11T19:05:03.484 DEBUG:teuthology.orchestra.run.smithi164:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph nfs cluster create test
2022-02-11T19:05:03.809 INFO:teuthology.orchestra.run.smithi164.stderr:Error EPERM: osd pool application enable failed: unrecognized pool '.nfs' retval: -2
2022-02-11T19:05:03.812 DEBUG:teuthology.orchestra.run:got remote process result: 1
2022-02-11T19:05:03.813 DEBUG:teuthology.orchestra.run.smithi164:> sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph log 'Ended test tasks.cephfs.test_nfs.TestNFS.test_cluster_info'
2022-02-11T19:05:04.089 INFO:journalctl@ceph.mgr.a.smithi164.stdout:Feb 11 19:05:03 smithi164 bash[32697]: debug 2022-02-11T19:05:03.807+0000 7f2f8585e700 -1 mgr.server reply reply (1) Operation not permitted osd pool application enable failed: unrecognized pool '.nfs' retval: -2
2022-02-11T19:05:04.090 INFO:journalctl@ceph.mon.a.smithi164.stdout:Feb 11 19:05:04 smithi164 bash[12809]: audit 2022-02-11T19:05:03.806384+0000 mgr.a (mgr.14302) 11 : audit [DBG] from='client.14318 -' entity='client.admin' cmd=[{"prefix": "nfs cluster create", "cluster_id": "test", "target": ["mon-mgr", ""]}]: dispatch
2022-02-11T19:05:04.090 INFO:journalctl@ceph.mon.a.smithi164.stdout:Feb 11 19:05:04 smithi164 bash[12809]: audit 2022-02-11T19:05:03.808598+0000 mon.a (mon.0) 411 : audit [INF] from='mgr.14302 172.21.15.164:0/2020663313' entity='mgr.a' cmd=[{"prefix": "osd pool create", "pool": ".nfs"}]: dispatch
2022-02-11T19:05:04.090 INFO:journalctl@ceph.mon.a.smithi164.stdout:Feb 11 19:05:04 smithi164 bash[12809]: audit 2022-02-11T19:05:03.810172+0000 mon.a (mon.0) 412 : audit [INF] from='mgr.14302 172.21.15.164:0/2020663313' entity='mgr.a' cmd=[{"prefix": "osd pool application enable", "pool": ".nfs", "app": "nfs"}]: dispatch
2022-02-11T19:05:05.078 INFO:tasks.cephfs_test_runner:test_cluster_info (tasks.cephfs.test_nfs.TestNFS) ... ERROR
``` 

http://pulpito.front.sepia.ceph.com/yuriw-2022-02-11_18:38:05-rados-wip-yuri-testing-2022-02-09-1607-distro-default-smithi/6677099/

**Comment by yuriw** [comment]
@Matan-B pls add `needs-qa` when comments addressed

**Comment by Matan-B** [comment]
@neha @yuriw Comments have been addressed.
Verified with `test_nfs` (3/4) - the failure has been fixed.
Although there is still one failed job but it seems unrelated since its pretty common.
https://pulpito.ceph.com/matan-2022-02-13_11:25:08-orch-master-distro-basic-smithi/

Latest master also fails the same test (other reason) with the same assertion:
```INFO:tasks.cephfs_test_runner:AssertionError: NFS Ganesha cluster deployment failed```
Perhaps we should increase `wait_time` in `_check_nfs_cluster_status` to allow ganesha daemon more time to be created/deleted.
https://pulpito.ceph.com/matan-2022-02-13_13:18:27-orch-master-distro-basic-smithi/

**Comment by neha-ojha** [comment]
> @Neha @yuriw Comments have been addressed. Verified with `test_nfs` (3/4) - the failure has been fixed. Although there is still one failed job but it seems unrelated since its pretty common. https://pulpito.ceph.com/matan-2022-02-13_11:25:08-orch-master-distro-basic-smithi/
> 
> Latest master also fails the same test (other reason) with the same assertion: `INFO:tasks.cephfs_test_runner:AssertionError: NFS Ganesha cluster deployment failed` Perhaps we should increase `wait_time` in `_check_nfs_cluster_status` to allow ganesha daemon more time to be created/deleted. https://pulpito.ceph.com/matan-2022-02-13_13:18:27-orch-master-distro-basic-smithi/

@adk3798 FYI

**Comment by adk3798** [comment]
> > @Neha @yuriw Comments have been addressed. Verified with `test_nfs` (3/4) - the failure has been fixed. Although there is still one failed job but it seems unrelated since its pretty common. https://pulpito.ceph.com/matan-2022-02-13_11:25:08-orch-master-distro-basic-smithi/
> > Latest master also fails the same test (other reason) with the same assertion: `INFO:tasks.cephfs_test_runner:AssertionError: NFS Ganesha cluster deployment failed` Perhaps we should increase `wait_time` in `_check_nfs_cluster_status` to allow ganesha daemon more time to be created/deleted. https://pulpito.ceph.com/matan-2022-02-13_13:18:27-orch-master-distro-basic-smithi/
> 
> @adk3798 FYI

Yeah, some of those nfs tests have been a bit inconsistent and I've been thinking about upping the timeouts a bit as well. I recently had a similar thing with test_orch_cli (https://github.com/ceph/ceph/pull/44965) where we increased the timeout and it made the tests go from occasional failures to (at least so far) always passing. To summarize, I think increasing the timeout is actually a fair idea. From my testing when these nfs tests did fail it tended to timeout only a few seconds before the thing we were waiting for came to be.

**Comment by ljflores** [comment]
http://pulpito.front.sepia.ceph.com/?branch=wip-yuri4-testing-2022-02-18-0800

Failures, unrelated:
    https://tracker.ceph.com/issues/45721
    https://tracker.ceph.com/issues/53422
    https://tracker.ceph.com/issues/51627
    https://tracker.ceph.com/issues/53680
    https://tracker.ceph.com/issues/52320
    https://tracker.ceph.com/issues/52124

Details:
    1. CommandFailedError: Command failed (workunit test rados/test_python.sh) FAIL: test_rados.TestWatchNotify.test - Ceph - RADOS
    2. tasks.cephfs.test_nfs.TestNFS.test_export_create_with_non_existing_fsname: AssertionError: NFS Ganesha cluster deployment failed - Ceph - Orchestrator
    3. FAILED ceph_assert(attrs || !recovery_state.get_pg_log().get_missing().is_missing(soid) || (it_objects != recovery_state.get_pg_log().get_log().objects.end() && it_objects->second->op == pg_log_entry_t::LOST_REVERT)) - Ceph - RADOS
    4. ERROR:tasks.rook:'waiting for service removal' reached maximum tries (90) after waiting for 900 seconds - Ceph - Orchestrator
    5. unable to get monitor info from DNS SRV with service name: ceph-mon - Ceph - Orchstrator
    6. Invalid read of size 8 in handle_recovery_delete() - Ceph - RADOS

**Pull request merged**